### PR TITLE
[docs] Clarify coverage-dependent surface species docs

### LINKED ISSUE #1579: Cantera 3.0 website coverage-dependent surface species documentation
**Problem description**

Documentation of [coverage-dependent-surface](https://cantera.org/documentation/dev/sphinx/html/yaml/species.html#coverage-dependent-surface) species thermo is confusing. 

image

Unlike other entries that (usually) correspond to a single C++ class/struct, this entry covers four different species coverage models (see [doxygen](https://cantera.org/documentation/dev/doxygen/html/db/d25/classCantera_1_1CoverageDependentSurfPhase.html#nested-classes) and [source code](https://github.com/Cantera/cantera/blob/4b30fa74bca55bccf476b826d26ff820a8a94ba4/include/cantera/thermo/CoverageDependentSurfPhase.h#L78-L124)). In #1548 (now closed), this was addressed by a separate entry "Species Coverage Models", see [proposed MarkDown](https://github.com/Cantera/cantera/pull/1548/files#diff-3657bd13dae0b3228be867ec8114a1a2646d6d73dff644205e970a1273c05741) (same level as "Species Critical State Parameters")

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

Clarify coverage-dependent surface species documentation

**If applicable, fill in the issue number this pull request is fixing**

Closes #1579

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: doc/sphinx/yaml/species.rst [hunk 1]
```diff
@@ -305,6 +458,23 @@ Example::
...
+
+Additional fields:
+
+``coverage-dependencies``
+    Mapping where keys are the name of species whose coverage affects
+    thermodynamic properties of the node-owner species. The map values are
+    the dependency entries including ``model`` and model-specific parameters as
+    described in :ref:`sec-yaml-species-coverage`.
+
+
```
Comment (speth) [inline doc/sphinx/yaml/species.rst]: This placement still implies that the ``coverage-dependencies`` field is a child of the ``equation-of-state`` field, but it's actually a top-level field within each species entry, i.e. alongside `thermo` and `equation-of-state`, and should be listed with the other fields at that level.

**Review by speth — CHANGES_REQUESTED** [review]
Thanks for looking into this @ischoegl. You're definitely right that this was organized incorrectly. I think there's just one remaining part that still needs to be moved to match the actual structure.

File: doc/sphinx/yaml/species.rst (hunk 1 above)
Comment (ischoegl) [reply doc/sphinx/yaml/species.rst]: Good catch!

**Comment by ischoegl** [comment]
Thanks for the feedback, @speth. This should be ready to merge.

File: doc/sphinx/yaml/species.rst (hunk 1 above)
Comment (speth) [reply doc/sphinx/yaml/species.rst]: Sorry, I'm having trouble describing what I mean. This is what I was trying to get across: https://github.com/speth/cantera/commit/db1a5d41b78a5675d8e2e8496961c1a33c53ec08.

Comment (ischoegl) [reply doc/sphinx/yaml/species.rst]: Ok. When I cherry-pick your commit, the only difference I see is that your version skips 

```
``coverage-dependencies``
    Mapping where keys are the name of species whose coverage affects
    thermodynamic properties of the node-owner species. The map values are
    the dependency entries including ``model`` and model-specific parameters
```

Is this the intent?

*PS: feel free to push suggestions to this PR directly*

Comment (speth) [reply doc/sphinx/yaml/species.rst]: It _should_ be that my branch moves that block up to the top, between the descriptions of the `transport` and `sites` fields. I'll just go ahead and push here.

**Review by speth — APPROVED** [review]

**Comment by speth** [comment]
@ischoegl, please feel free to merge if this updated version makes sense to you.

**Pull request merged**